src/components/events/CalendarButtons/index.tsx

```diff
@@ -54,31 +52,45 @@ const saveToAppleCal = (event: PublicEvent): void => {
 
   const response = ics.createEvent(attributes);
   if (response.value) {
-    createDownloadFile(response.value, formatURLEventTitle(event.title));
-  } else {
-    showToast(response.error?.message || 'An error occurred when saving to Apple Calendar.');
+    // Referencing this answer to save text in a file and download it
+    // https://stackoverflow.com/questions/44656610/download-a-string-as-txt-file-in-react
+    const file = new Blob([response.value], { type: 'text/calendar' });
+    const href = URL.createObjectURL(file);
+    const download = `${formatURLEventTitle(event.title)}.ics`;
```

Comment:
you should probably put this in a `useMemo` or something. it's probably not a big deal because the ics file probably isn't that big, and there aren't that many interactive elements on the page, but every time React re-renders, it will create a new object url, which would lead to a memory leak

Reply:
I've been playing around with this for like an hour and for some reason this approach isn't really working.

It seems like the blob/objectURL no longer works after refreshing the page, so after hitting refresh none of the apple calendar buttons work.

I'm unsure if this is a local-only problem because the console error after hitting the apple calendar link after a refresh is `Not allowed to load local resource: blob:nodedata:...`

I'm going to push what I have sans the `useMemo` for appleCal and we can revisit in the future